application/src/main/java/org/thingsboard/server/service/edge/rpc/EdgeGrpcSession.java

```diff
@@ -198,29 +187,72 @@ private void closeSession() {
                 if (edge != null) {
                     try {
                         sessionCloseListener.accept(edge.getId());
-                    } catch (Exception ignored) {}
+                    } catch (Exception ignored) {
+                    }
                 }
                 try {
                     outputStream.onCompleted();
-                } catch (Exception ignored) {}
+                } catch (Exception ignored) {
+                }
             }
         };
     }
 
+    public void startSyncProcess(TenantId tenantId, EdgeId edgeId) {
+        log.trace("[{}][{}] Staring edge sync process", tenantId, edgeId);
+        syncCompleted = false;
+        syncExecutorService.submit(() -> {
```

Comment:
what is the purpose of the sync? You do not guarantee that startSyncProcess will not run in parallel for the same edge.

Reply:
Maybe define some timeout for edge sync?